cardinal/plugin_persona.go

```diff
@@ -102,15 +102,25 @@ func (p *personaPlugin) RegisterMessages(world *World) error {
 // users who want to interact with the game via smart contract can link their EVM address to their persona tag, enabling
 // them to mutate their owned state from the context of the EVM.
 func AuthorizePersonaAddressSystem(wCtx engine.Context) error {
-	personaTagToAddress, err := buildPersonaIndex(wCtx)
-	if err != nil {
-		return err
-	}
+	// The persona tag index will be lazily built for the first AuthorizePersonaAddress message. If no such messages
+	// are present, the index will not be built.
+	var personaTagToAddress personaIndex
```

Comment:
Wouldn't it be better if we have an index that persist across ticks?

Reply:
I'd say that we have to do this because otherwise we would have a tick drift everytime someone interacts with Persona [](edited-with-Axolo)

Reply:
At least until we have a proper way to do indexing [](edited-with-Axolo)

Reply:
the concept of an index will always be "out of ecs" until we change storage. That's where an index would typically go. I agree with scott here, a bit of a temp hack is ok [](edited-with-Axolo)

Reply:
OK, persona tag tracking has been moved to an in-memory var. 

the `tickOfPersonaTagToAddressIndex` is used to tell if we're in a test so the index can be rebuilt. This seemed more straight forward than creating a "clear index" public function or trying to use an init function. If either of you have an alternative suggestion I'm all ears.